crates/common/mqtt_channel/src/config.rs

```diff
@@ -0,0 +1,121 @@
+use crate::TopicFilter;
+
+/// Configuration of an MQTT connection
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// MQTT host to connect to
+    ///
+    /// Default: "localhost"
+    pub host: String,
+
+    /// MQTT port to connect to
+    ///
+    /// Default: 1883
+    pub port: u16,
+
+    /// The session name to be use on connect
+    ///
+    /// If no session name is provided, a random one will be created on connect.
+    ///
+    /// Default: None
+    pub session_name: Option<String>,
+
+    /// The list of topics to subscribe to on connect
+    ///
+    /// Default: An empty topic list
+    pub subscriptions: TopicFilter,
+
+    /// Clean the MQTT session upon connect if set to `true`.
+    ///
+    /// Default: `false`.
+    pub clean_session: bool,
+
+    /// Capacity of the internal message queues
+    ///
+    /// Default: `1024`.
+    ///
+    pub queue_capacity: usize,
+
+    /// Maximum size for a message payload
+    ///
+    /// Default: `8 * 1024`.
+    pub max_packet_size: usize,
+}
+
+/// By default a client connects the local MQTT broker.
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            host: String::from("localhost"),
+            port: 1883,
+            session_name: None,
+            subscriptions: TopicFilter::empty(),
+            clean_session: false,
+            queue_capacity: 1024,
+            max_packet_size: 8 * 1024,
```

Comment:
Correct me if I'm wrong but this will make the max packet size 8KB, isn't this a little small? I think we may be aiming in something more like 8MB?

On the other hand this is an ok default, but maybe a mention of this in documentation would be a nice to have?

Reply:
Set to 1M and documented.